app/models/home.rb

```diff
@@ -4,6 +4,15 @@ class Home < ActiveRecord::Base
   validates :price_per_night, presence: true
   validates :address,         presence: true
   belongs_to :user
+  has_many :reservations
   has_many :photos, :dependent => :destroy, inverse_of: :home
   accepts_nested_attributes_for :photos, :allow_destroy => true
+
+  def upcoming_reservations
+    self.reservations.where("checkout > ?", Date.today)
+  end
+
```

Comment:
this is really clean code!

two thoughts for future reference:
1) we can dry out this and the same in the user mode
2) we need to test the edge cases for reservations on the same day - your solution is way better than what we had in toolchest though